app/src/main/java/net/squanchy/schedule/ScheduleService.kt

```diff
@@ -1,88 +1,100 @@
 package net.squanchy.schedule
 
 import io.reactivex.Observable
-import io.reactivex.functions.Function3
-import net.squanchy.schedule.domain.view.Event
+import io.reactivex.functions.BiFunction
+import io.reactivex.schedulers.Schedulers
 import net.squanchy.schedule.domain.view.Schedule
 import net.squanchy.schedule.domain.view.SchedulePage
 import net.squanchy.schedule.domain.view.Track
+import net.squanchy.schedule.filterschedule.TracksFilter
 import net.squanchy.service.firebase.FirebaseAuthService
 import net.squanchy.service.firestore.FirestoreDbService
+import net.squanchy.service.firestore.model.schedule.FirestoreEvent
 import net.squanchy.service.firestore.model.schedule.FirestoreSchedulePage
 import net.squanchy.service.firestore.toEvent
-import net.squanchy.service.firestore.toTrack
-import net.squanchy.service.repository.FilterScheduleRepository
 import net.squanchy.support.lang.Checksum
-import net.squanchy.support.lang.Optional
 import org.joda.time.DateTimeZone
 import org.joda.time.LocalDate
 
 interface ScheduleService {
-    fun schedule(onlyFavorites: Boolean): Observable<Schedule>
+    fun schedule(onlyFavorites: Boolean = false): Observable<Schedule>
     fun currentUserIsSignedIn(): Observable<Boolean>
 }
 
 class FirestoreScheduleService(
     private val authService: FirebaseAuthService,
     private val dbService: FirestoreDbService,
-    private val filterScheduleRepository: FilterScheduleRepository,
+    private val tracksFilter: TracksFilter,
     private val checksum: Checksum
 ) : ScheduleService {
 
     override fun schedule(onlyFavorites: Boolean): Observable<Schedule> {
-        return Observable.combineLatest(dbService.scheduleView(), dbService.timezone(), filterScheduleRepository.filters, combineInATriple())
-            .map { scheduleViewTimeZoneAndFilters ->
-                val (scheduleView, timeZone, filters) = scheduleViewTimeZoneAndFilters
-                extractTracksFrom(scheduleView)
-                val schedulePages = createSchedulePages(scheduleView, timeZone, onlyFavorites, filters, checksum)
+        val filteredDbSchedulePages = dbService.scheduleView()
+            .filterByFavorites(onlyFavorites)
+            .filterByTracks(tracksFilter.selectedTracks)
+
+        val domainSchedulePages = Observable.combineLatest(
+            filteredDbSchedulePages,
+            dbService.timezone(),
+            toSortedDomainSchedulePages()
+        )
+
+        return Observable.combineLatest(domainSchedulePages, dbService.timezone(), combineInAPair())
```

Comment:
what if instead of combining them in a pair, we combine them directly in a schedule (since we don't do anything else anyways)? At that point I think you could pass a reference to the constructor

Reply:
940d982630ef1850142e12050746762cddd5b903 together with previous comment